keps/sig-apps/2599-minreadyseconds-for-statefulsets/README.md

```diff
@@ -0,0 +1,598 @@
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The main goal of this enhancement is to introduce minReadySeconds as an 
+optional field to StatefulSets. 
+
+## Motivation
+
+minReadySeconds specifies the minimum number of seconds for which a newly 
+created Pod should be ready without any of its containers crashing, for it 
+to be considered available. This defaults to 0 (the pod will be considered 
+available as soon as it is ready).
+
+minReadySeconds is already available as an optional field for Deployments,
+DaemonSets. Enabling this option helps in bringing StatefulSets on par 
+with other workload controllers.
+
+### Goals
+
+StatefulSet Controller honoring minReadySeconds and mark Pod ready only if Pod 
+is available for given time mentioned in minReadySeconds.
+
+### Non-Goals
+
+Moving minReadySeconds to Pod Spec is beyond the scope of this KEP
```

Comment:
I would describe why, and maybe say a future story could address it based on consistent use.  First, the effort to change pod spec would be large (impact).  Second, currently our workload controllers are inconsistent and we prioritize consistent of experience over desirable future state (consistency).  Third, stateful sets are just different enough from daemonsets and deployments that real world use of minReadySeconds for stateful sets might influence any future design or point in a more appropriate direction (completeness)